_posts/2021-11-08-changelog-159.markdown

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #159"
+date: 2021-11-08 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#7914], [#8029] Use info calculated by new name resolution engine for `Import` quick-fix and completion of out-of-scope items. It brings:
+    - Better processing of complex re-exports, including proper visibility handling
+    - Significant performance improvement
+    - Auto-import for macros
+
+      {% include gif-img.html path="/assets/posts/changelog-159/auto_import" w="550px" %}
+
+
+* [#7798] Support [let else](https://rust-lang.github.io/rfcs/3137-let-else.html) syntax
+
+  {% include gif-img.html path="/assets/posts/changelog-159/let_else" w="550px" %}
+
+* [#7898] Provide completion of the [`rust-version`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field) field in the `[package]` table of the `Cargo.toml` manifest (by [@HTG-YT])
+
+  <img src="/assets/posts/changelog-159/rust_version.png" alt="rust_version completion" width="550px">
+
+* [#7928] Enable `Unresolved reference` inspection for attribute macros if `org.rust.macros.proc` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) is enabled
+
+## Performance Improvements
+
+* [#7944] Slightly optimize `Find usages` for public items in binary crates
+
+* [#7936] Slightly optimize memory consumption of name resolution internal structures
+
+## Fixes
+
+* [#8033] Fix name resolution of explicit type-qualified associated types in type aliases (like `type Alias<T> = <T as Trait>::Item;`). This fixes type inference in [`hecs`](https://crates.io/crates/hecs) library
```

Comment:
It's not only about `hecs` lib. Let's rephrase it somehow. `hecs` can be used as an example